code/script.js

```diff
@@ -247,97 +252,175 @@ const generateBoard = () => {
   // This variable stores what option group (category) the question belongs to.
+  const value = questions.options[questions.selectedIndex].value // i added this here
   // We also need a variable that stores the actual value of the question we've selected.
 
   if (category === 'hair color') {
     currentQuestion = {
       attribute: 'hairColor',
-      // value: ,
+      value,
       // 👆 add the value from the input here
-      category: category,
+      category
     }
   } else if (category === 'eye color') {
+    currentQuestion = {
+      attribute: 'eyeColor',
+      value,
+      category
+    }
     // Set this up your self
   } else if (category === 'accessories') {
     currentQuestion = {
-      //attribute: ,
+      attribute: value, // do something here!!!
       // 👆 this is the property of the booleans such as smoke, glasses and hat. add the value from the input here
       value: true, // we're asking if this person wears a hat for exaple, so always true in the question.
-      category: category,
+      category
     }
   } else if (category === 'other') {
+    currentQuestion = {
+      value: true,
+      category
+    }
     // Set this up your self (should be same structure as above)
   }
+  console.log(value) // i added this here
 }
 
 // This function should be invoked when you click on 'Find Out'.
 const checkQuestion = () => {
+  let keep
+  let secretValue = secret[currentQuestion.attribute]
+  if (currentQuestion.value === secretValue) {
+    keep = true
+  } else {
+    keep = false
+  }
+
+  filterCharacters(keep)
   // Compare the currentQuestion with the secret person.
   // See if we should keep or remove people based on that
   // Then invoke filterCharacters
 }
 
 // It'll filter the characters array and redraw the game board.
 const filterCharacters = (keep) => {
+  const category = questions.options[questions.selectedIndex].parentNode.label
+  console.log(category)
   // Show the correct alert message for different categories
-  if (group === 'accessories') {
+  if (category === 'accessories') {
     if (keep) {
       alert(
-        `Yes, the person wears ${attribute}! Keep all that wears ${attribute}`
+        `Yes, the person wears ${[currentQuestion.attribute]}! Keep all that wears ${[currentQuestion.attribute]}`
       )
+      charactersInPlay = charactersInPlay.filter((person) => person[currentQuestion.attribute] === currentQuestion.value)
     } else {
       alert(
-        `No, the person doesn't wear ${attribute}! Remove all that wears ${attribute}`
+        `No, the person doesn't wear ${[currentQuestion.attribute]}! Remove all that wears ${[currentQuestion.attribute]}`
       )
+      charactersInPlay = charactersInPlay.filter((person) => person[currentQuestion.attribute] !== currentQuestion.value) 
     }
-  } else if (group === 'other') {
-    // Similar to the one above
-  } else {
+  } else if (category === 'other') {
```

Comment:
Smoking filtering is not working. One reason is that you haven´t defined "attribute: actualValue " property (see my comment in line 298). And here similar approach as for other categories´ alert messages should be used: 
 
else if (category === 'other') {
 if (keep) {
      alert(
        `Yes, the person is a ${currentQuestion.attribute}! Keep all persons who are ${attribute}`
 )
    } else {
      alert(
        `No, the person is not a ${currentQuestion.attribute}! Remove all persons who are not ${attribute}`
      )
    }